src/BaseActionWatcher.ts

```diff
@@ -28,26 +37,74 @@ export class BaseActionWatcher {
    * Start a polling loop
    */
   public async watch(isReplay: boolean = false) {
-    const startTime = new Date().getTime()
+    try {
+      this.running = true
+      const startTime = new Date().getTime()
+
+      await this.checkForBlocks(isReplay)
+      if (this.shouldPause) {
+        this.running = false
+        this.shouldPause = false
+        this.log.info("Demux paused.")
+        return
+      }
+
+      const endTime = new Date().getTime()
+      const duration = endTime - startTime
+      let waitTime = this.pollInterval - duration
+      if (waitTime < 0) {
+        waitTime = 0
+      }
+
+      setTimeout(async () => await this.watch(false), waitTime)
+    } catch (err) {
+      this.running = false
+      this.shouldPause = false
+      this.log.error(err)
+      this.log.info("Demux stopped due to error.")
+    }
+  }
 
-    await this.checkForBlocks(isReplay)
+  public start() {
+    if (this.running) {
+      return false
+    }
+    this.log.info("Demux starting...")
+    this.watch()
+    return true
+  }
 
-    const endTime = new Date().getTime()
-    const duration = endTime - startTime
-    let waitTime = this.pollInterval - duration
-    if (waitTime < 0) {
-      waitTime = 0
+  public pause() {
+    if (!this.running) {
+      return false
```

Comment:
do we want contextual logging here

```this.log.info("Demux is not currently running")```

Reply:
I was planning on putting these logs in the Express routes, but I think you're right that it's actually better here

Reply:
I usually try to avoid anything app specific in routes.  Routes are an entry point, but otherwise, they should be dumb.